var/spack/repos/builtin/packages/zlib/package.py

```diff
@@ -50,12 +50,30 @@ class Zlib(Package):
 
     patch('w_patch.patch', when="@1.2.11%cce")
 
+    @property
+    def headers(self):
+        # If zlib is configured as external package, e.g. in /usr, searching the
+        # whole prefix.include recursively could be slow and returns a ton of
+        # headers, so we override this default behavior.
+        hdr = find_headers('zlib', root=self.prefix.include, recursive=False)
+        return hdr or None
```

Comment:
i would just return `find_headers`, why would you need `None`? IMO `find_headers` should actually throw error if it can't find a header, which means something is broken.

Reply:
Returning `None` will throw an `AttributeError` with a nicely formatted message, so for zlib it makes sense to return `None` if the header is not found. I'm not sure about modifying `find_headers` to throw when nothing is found - there maybe cases where that is not actually an error.

Reply:
ah, thanks for explanation, I did not know that.

Reply:
Looking again at the `AttributeError` message triggered by returning `None`:
https://github.com/spack/spack/blob/819ceb4f6dc0bd6a3b68df28f54f19ca0ad368ae/lib/spack/spack/spec.py#L832-L842
I'm not so sure it was really meant for the case of "library/header file not found".

Reply:
I proposed a solution (PR #7277) that should work better when returning `None` from properties like this.